src/simtools/visualization/legend_handlers.py

```diff
@@ -141,248 +96,228 @@ class MeanRadiusOuterEdgeObject:
 
 
-class SquarePixelHandler:
-    """Legend handler class to plot a square "on" pixel."""
+class _BaseSquarePixelHandler:
+    """Base class for square pixel handlers."""
 
     @staticmethod
-    def legend_artist(_, __, ___, handlebox):  # noqa: D102
+    def _create_square_patch(handlebox, facecolor, edgecolor):
+        """Create a square patch with specified colors."""
         x0, y0 = handlebox.xdescent, handlebox.ydescent
         width = height = handlebox.height
         patch = mpatches.Rectangle(
             [x0, y0],
             width,
             height,
-            facecolor=(1, 1, 1, 0),
-            edgecolor=(0, 0, 0, 1),
+            facecolor=facecolor,
+            edgecolor=edgecolor,
             transform=handlebox.get_transform(),
         )
         handlebox.add_artist(patch)
         return patch
 
 
-class SquareEdgePixelHandler:
+class SquarePixelHandler(_BaseSquarePixelHandler):
+    """Legend handler class to plot a square "on" pixel."""
+
+    @staticmethod
+    def legend_artist(_, __, ___, handlebox):
+        """Legend artist."""
+        return SquarePixelHandler._create_square_patch(
+            handlebox, facecolor=(1, 1, 1, 0), edgecolor=(0, 0, 0, 1)
+        )
+
+
+class SquareEdgePixelHandler(_BaseSquarePixelHandler):
     """Legend handler class to plot a square "edge" pixel."""
 
     @staticmethod
-    def legend_artist(_, __, ___, handlebox):  # noqa: D102
-        x0, y0 = handlebox.xdescent, handlebox.ydescent
-        width = height = handlebox.height
-        patch = mpatches.Rectangle(
-            [x0, y0],
-            width,
-            height,
+    def legend_artist(_, __, ___, handlebox):
+        """Legend artist."""
+        return SquareEdgePixelHandler._create_square_patch(
+            handlebox,
             facecolor=(*mcolors.to_rgb("brown"), 0.5),
             edgecolor=(*mcolors.to_rgb("black"), 1),
-            transform=handlebox.get_transform(),
         )
-        handlebox.add_artist(patch)
-        return patch
 
 
-class SquareOffPixelHandler:
+class SquareOffPixelHandler(_BaseSquarePixelHandler):
     """Legend handler class to plot a square "off" pixel."""
 
     @staticmethod
-    def legend_artist(_, __, ___, handlebox):  # noqa: D102
-        x0, y0 = handlebox.xdescent, handlebox.ydescent
-        width = height = handlebox.height
-        patch = mpatches.Rectangle(
-            [x0, y0],
-            width,
-            height,
-            facecolor="black",
-            edgecolor="black",
-            transform=handlebox.get_transform(),
+    def legend_artist(_, __, ___, handlebox):
+        """Legend artist."""
+        return SquareOffPixelHandler._create_square_patch(
+            handlebox, facecolor="black", edgecolor="black"
         )
-        handlebox.add_artist(patch)
-        return patch
 
 
-class LSTHandler(TelescopeHandler):
-    """Legend handler class to plot a representation of an LST in an array layout."""
+class TelescopeHandler:
```

Comment:
This class name and docstring is a bit undescriptive. Maybe TelescopeGeometryArtist or similar?